src/map_field.cpp

```diff
@@ -413,6 +413,8 @@ bool map::process_fields_in_submap( submap *const current_submap,
                     debugmsg( "Whoooooa intensity of %d", cur.get_field_intensity() );
                 }
 
+                dirty_transparency_cache = curtype.obj().dirty_transparency_cache;
```

Comment:
The logic here is wrong, it needs to "latch" to dirty. Right now it will be un-set when it processes an invisible field.
```suggestion
                dirty_transparency_cache |= curtype.obj().dirty_transparency_cache;
```
I think this will do it, however when I test the smoke from a smoke grenade still isn't acting non-transparent.

Reply:
Re-tested with current master and it's already broken, so that's great :/

Reply:
Maybe be I broke it earlier? Need to test in 0.D...

Reply:
0.D is working, so I need to bisect.

Reply:
Smoke is non-transparent starting from intensity level 2.

Maybe it is not transparency, but field intensity issue (i.e. smoke from smoke bombs aren't achieving level 2 or 3 anymore)?

I've tested on savegame from 0.D with exploded smoke bomb and there is thick smoke in the center which is non-transparent, but newly exploded smoke bomb only gives intensity level 1.

Reply:
Should be fixed in https://github.com/CleverRaven/Cataclysm-DDA/pull/32551.